text/0000-rand-crypto.md

```diff
@@ -0,0 +1,192 @@
+- Feature Name: rand-crypto
+- Start Date: 2017-08-18
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Background
+
+This RFC is an offshoot of the [Rand revision RFC], on the subject of:
+
+*   splitting rand into multiple crates
+*   splitting the current `Rng` trait into a crypto-oriented trait and
+    numerical-application-oriented trait
+*   impl of above trait(s) for OS-provided randomness
+*   naming conventions for above things
+
+# Summary
+[summary]: #summary
+
+Create a new crate, `crypto-rand`, with only the following contents:
+
+```rust
+pub enum Error {
+    pub Unspecified,
+}
+
+pub trait Generator {
+    fn try_fill(&self, dest: &mut [u8]) -> Result<(), Error>;
+}
+```
+
+Associated with this, create another crate called `crypto-rng-os` which defines
+a struct `OsGen`, implementing `Generator` via OS functionality (using code from
+[rand/os.rs](https://github.com/rust-lang-nursery/rand/blob/master/src/os.rs)
+and/or
+[ring/rand.rs](https://github.com/briansmith/ring/blob/master/src/rand.rs).
+
+This RFC does not directly propose adding other implementations of `Generator`,
+but if any are written to be shared, they can be placed in a small crate named
+`crypto-rand-NAME`.
+
+All crates should be adopted into
+[rust-lang-nursery](https://github.com/rust-lang-nursery/) or (eventually)
+[rust-lang](https://github.com/rust-lang), or another community-maintained
+collection of repositories. This should prevent crates from becoming orphaned.
+
+Finally, add an impl for `rand::Rng` of `crypto_rand::Generator` so that the
+`rand` crate can use this `OsGen`. (Alternatively, `rand` could create a wrapper
+`OsRng` maintaining the current functionality. `rand::Rng` may get renamed.
+This is beyond the scope of this RFC.)
+
+`rand::Rng` is not the subject of this RFC, but it is assumed here that this
+trait will have an API like the following:
+
+```rust
+pub trait Rng {
+    fn next_u32(&mut self) -> u32;
+    fn next_u64(&mut self) -> u64 { /* default impl here */ }
+    // possibly also next_u128
+    
+    // possibly also the following (no Result unlike try_fill above):
+    fn fill(&mut self, dest: &mut [u8]) { /* default impl here */ }
```

Comment:
You need to specify what will happen here in case of an error, will it panic? Ideally directly provide default implementation for this method.